processing/src/test/java/org/apache/druid/common/semantic/SemanticCreatorUsageTest.java

```diff
@@ -87,8 +87,8 @@ public void testReturnType()
   {
     Class<?> returnType = method.getReturnType();
     assertTrue(
-        returnType + " is not an interface; this method must return with an interface; ",
-        returnType.isInterface()
+        returnType + " is not an interface or abstract class; this method must return with either of them; ",
+        returnType.isInterface() || Modifier.isAbstract(returnType.getModifiers())
```

Comment:
oh I see; `FrameRowsAndColumns` is an abstract class - however you need the `getSignature` on the interface ....

can we promote `FrameRowsAndColumns` to an interface ; and make the current one `AbstractFrameRowsAndColumns` ?

Reply:
I guess that would be more cleaner. Will update it.